daemon/state.go

```diff
@@ -325,26 +325,27 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) {
 }
 
 func (d *Daemon) allocateIPsLocked(ep *endpoint.Endpoint) error {
-	err := ipam.AllocateIP(ep.IPv6.IP())
-	if err != nil {
-		// TODO if allocation failed reallocate a new IP address and setup veth
-		// pair accordingly
-		return fmt.Errorf("unable to reallocate IPv6 address: %s", err)
-	}
+	var err error
 
-	defer func(ep *endpoint.Endpoint) {
+	if option.Config.EnableIPv6 && ep.IPv6 != nil {
```

Comment:
If the endpoint didn't contain an IPv6 shouldn't we try to assign a new one now?

Reply:
That's potentially quite dangerous as the application inside the container is already running. Assigning an IPv6 address in the middle may break things. Just like we keep using addresses when disabling an address family, we should probably not add a new address to already existing containers.